test/integration/crud/find.test.js

```diff
@@ -786,83 +791,96 @@ describe('Find', function () {
 
     test: function (done) {
       var configuration = this.configuration;
-      var client = configuration.newClient(configuration.writeConcernMax(), { maxPoolSize: 1 });
-      client.connect(function (err, client) {
-        var db = client.db(configuration.db);
-        db.createCollection('test_find_and_modify_a_document_1', function (err, collection) {
-          // Test return new document on change
-          collection.insert({ a: 1, b: 2 }, configuration.writeConcernMax(), function (err) {
+      var db = client.db(configuration.db);
+      db.dropCollection('test_find_and_modify_a_document_1')
+        .catch(() => null)
+        .finally(() => {
+          db.createCollection('test_find_and_modify_a_document_1', function (err, collection) {
             expect(err).to.not.exist;
 
-            // Let's modify the document in place
-            collection.findOneAndUpdate(
-              { a: 1 },
-              { $set: { b: 3 } },
-              { returnDocument: ReturnDocument.AFTER },
-              function (err, updated_doc) {
-                test.equal(1, updated_doc.value.a);
-                test.equal(3, updated_doc.value.b);
+            // Test return new document on change
+            collection.insert({ a: 1, b: 2 }, configuration.writeConcernMax(), function (err) {
```

Comment:
Since we're trying to get rid of instances of `collection.insert`, why don't we get rid of these instances when updating old tests as part of our drive-by cleanup?

Reply:
The change to this test was caused by unindenting the connect callback to fix a leak that showed up when it was erroring. There's a lot more cleanup that could be done here, remove callbacks, remove the old topology filtering, remove the combination of operations (dropColl, createColl, insert should all not be included in a test focused on findOneAndUpdate) and even fix the test name from being all camelcase. But there's only so much drive-by you can do before you're overloading the original task. 

Another reason I leave all of above mentioned legacy practices behind is that it earmarks the test for those of us who've been working on the code for awhile. When this test causes the next engineer some trouble, they'll know it isn't a recent addition by all these other outstanding issues with it. My drive-by goal was to address the leak issue, the next one may be to address any of the above issues when it causes trouble again.